testing/test_variable_formatting.py

```diff
@@ -0,0 +1,66 @@
+import ast
+import re
+from pathlib import Path
+from typing import List, Tuple
+
+import pytest
+from flake8_annotations import checker
+
+TEST_FILE = Path("./testing/code/variable_formatting.py")
+ERROR_CODE_TYPE = Tuple[int, int, str, checker.TypeHintChecker]
+SIMPLE_ERROR_CODE = Tuple[str, str]
+
+# Error type specific matching patterns
+TEST_ARG_NAMES = {"TYP001": "some_arg", "TYP002": "some_args", "TYP003": "some_kwargs"}
+RE_DICT = {"TYP001": r"'(\w+)'", "TYP002": r"\*(\w+)", "TYP003": r"\*\*(\w+)"}
+
+
+def _simplify_error(error_code: ERROR_CODE_TYPE) -> SIMPLE_ERROR_CODE:
+    """
+    Simplify the error yielded by the flake8 checker into an (error type, argument name) tuple.
+
+    Input error codes are assumed to be tuples of the form:
+    (line number, column number, error string, checker class)
+
+    Where the error string begins with "TYPxxx" and contains the arg name in the following form:
+    TYP001: '{arg name}'
+    TYP002: *{arg name}
+    TYP003: **{arg name}
+    """
+    error_type = error_code[2].split()[0]
+    arg_name = re.findall(RE_DICT[error_type], error_code[2])[0]
+    return error_type, arg_name
+
+
+class TestArgumentFormatting:
+    """Testing class for containerizing parsed error codes & running the fixtured tests."""
+
+    with TEST_FILE.open("r") as f:
+        src = f.read()
+
+    tree = ast.parse(src)
+    lines = src.splitlines()
+
+    error_codes = [_simplify_error(error) for error in checker.TypeHintChecker(tree, lines).run()]
+
+    batched_error_codes = {error_type: [] for error_type in TEST_ARG_NAMES.keys()}
+    for code, arg_name in error_codes:
+        batched_error_codes[code].append(arg_name)
```

Comment:
For some reason I found this really hard to wrap my head around, as in what the code does as a whole. Yet individually it doesn't seem unreadable by any means. Anyway, I think this is clearer:

```py
from collections import defaultdict

batched_error_codes = defaultdict(list)

for error in checker.TypeHintChecker(tree, lines).run():
    code, arg_name = _simplify_error(error)
    batched_error_codes[code].append(arg_name)
```

If you're not a fan of importing `defaultdict` then `setdefault()` is an alternative.